Api/src/guild-message/guild-message.resolver.ts

```diff
@@ -1,35 +1,60 @@
-import { Resolver, Query, Mutation, Args, Int } from '@nestjs/graphql';
+import {
+  Args,
+  Int,
+  Mutation,
+  Parent,
+  Query,
+  ResolveField,
+  Resolver,
+} from '@nestjs/graphql';
 import { GuildMessageService } from './guild-message.service';
 import { GuildMessage } from './entities/guild-message.entity';
 import { CreateGuildMessageInput } from './dto/create-guild-message.input';
 import { UpdateGuildMessageInput } from './dto/update-guild-message.input';
+import { Guild } from '../guild/entities/guild.entity';
 
 @Resolver(() => GuildMessage)
 export class GuildMessageResolver {
   constructor(private readonly guildMessageService: GuildMessageService) {}
 
   @Mutation(() => GuildMessage)
-  createGuildMessage(@Args('createGuildMessageInput') createGuildMessageInput: CreateGuildMessageInput) {
+  createGuildMessage(
+    @Args('createGuildMessageInput')
+    createGuildMessageInput: CreateGuildMessageInput,
+  ): Promise<GuildMessage> {
     return this.guildMessageService.create(createGuildMessageInput);
   }
 
   @Query(() => [GuildMessage], { name: 'guildMessage' })
-  findAll() {
+  findAll(): Promise<GuildMessage[]> {
     return this.guildMessageService.findAll();
   }
 
   @Query(() => GuildMessage, { name: 'guildMessage' })
-  findOne(@Args('id', { type: () => Int }) id: number) {
+  findOne(@Args('id', { type: () => Int }) id: number): Promise<GuildMessage> {
     return this.guildMessageService.findOne(id);
   }
 
+  @ResolveField(() => Guild)
+  getGuild(@Parent() guildMessage: GuildMessage): Promise<Guild> {
+    return this.guildMessageService.getGuild(guildMessage.guild_id);
+  }
+
   @Mutation(() => GuildMessage)
-  updateGuildMessage(@Args('updateGuildMessageInput') updateGuildMessageInput: UpdateGuildMessageInput) {
-    return this.guildMessageService.update(updateGuildMessageInput.id, updateGuildMessageInput);
+  updateGuildMessage(
+    @Args('updateGuildMessageInput')
+    updateGuildMessageInput: UpdateGuildMessageInput,
+  ): Promise<GuildMessage> | null {
+    return this.guildMessageService.update(
+      updateGuildMessageInput.id,
+      updateGuildMessageInput,
+    );
   }
 
   @Mutation(() => GuildMessage)
-  removeGuildMessage(@Args('id', { type: () => Int }) id: number) {
+  removeGuildMessage(
+    @Args('id', { type: () => Int }) id: number,
+  ): Promise<GuildMessage> | null {
     return this.guildMessageService.remove(id);
```

Comment:
Check the diffrence between remove and delete, and if we should use one or the other. I always used delete wich gives a delete result :D

Reply:
I would suggest doing this in another pull request, so you can rebase from dev :D.
I'll now merge and resolve this comment.